Client/Frontend/Browser/HandoffManager.swift

```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+class HandoffManager: NSObject {
+    static var sharedInstance = HandoffManager()
+    
+    lazy var userActivity: NSUserActivity? = NSUserActivity(activityType: NSUserActivityTypeBrowsingWeb)
```

Comment:
When using a lazy var you actually don't need to declare it as an optional if you wrap the result in a block. For example you can change this to:

```swift
lazy var userActivity: NSUserActivity = {
    return NSUserActivity(activityType: NSUserActivityTypeBrowsingWeb)
}()
```
Now you don't need to sprinkle questions marks throughout!